src/io.mli

```diff
@@ -30,4 +30,10 @@ module type S = sig
   val close : t -> unit
 
   val valid_fd : t -> bool
+
+  type fd
+
+  val lock : string -> fd
```

Comment:
w.r.t making this IO signature Mirage-compatible in future, I wonder if something like:
```ocaml
val with_lock: (unit -> 'a) -> 'a
```
might be a less Unix-ish abstraction. Plus, manual lock handling makes me nervous. (any thoughts, @dinosaure ?)

Reply:
(obviously, the manual `lock`/`unlock` functions are perfectly OK. Just conjecturing)

Reply:
I think in that case it's a bit difficult to do differently as the lock has to hold for the duration of the program.

Reply:
Besides this, the type name doesn't need to leak information about the implementation, maybe `type lock` would be easier to understand in the API ?

Reply:
I can confirm that a `type lock` should help a lot for a Mirage compat', like:
```ocaml
type 'a io
type lock

val create : unit -> lock
val lock : lock -> unit io
val unlock : lock -> unit
```